core/controller/src/main/scala/whisk/core/controller/Meta.scala

```diff
@@ -115,67 +224,216 @@ trait WhiskMetaApi extends Directives with PostActionActivation {
-                        "__ow_meta_verb" -> method.value.toLowerCase.toJson,
-                        "__ow_meta_path" -> restofPath.toJson,
-                        "__ow_meta_namespace" -> user.namespace.toJson)
-                    invokeAction(identity, action, Some(JsObject(content)), blocking = true, waitOverride = Some(WhiskActionsApi.maxWaitForBlockingActivation))
+    private def processAnonymousRequest(actionName: FullyQualifiedEntityName, context: Context, responseType: String)(
+        implicit transid: TransactionId) = {
+        // checks that subject has right to post an activation and fetch the action
+        // followed by the package and merge parameters. The action is fetched first since
+        // it will not succeed for references relative to a binding, and the export bit is
+        // confirmed before fetching the package and merging parameters.
+        def precheck: Future[(Identity, WhiskAction)] = for {
+            // lookup the identity for the action namespace
+            identity <- identityLookup(actionName.path.root) flatMap {
+                i => entitlementProvider.checkThrottles(i) map (_ => i)
+            }
+
+            // lookup the action - since actions are stored relative to package name
+            // the lookup will fail if the package name for the action refers to a binding instead
+            action <- confirmExportedAction(actionLookup(actionName, failureCode = NotFound)) flatMap { a =>
+                if (a.namespace.defaultPackage) {
+                    Future.successful(a)
                 } else {
-                    Future.failed(RejectRequest(BadRequest, Messages.parametersNotAllowed))
+                    pkgLookup(a.namespace.toFullyQualifiedEntityName) map {
+                        pkg => (a.inherit(pkg.parameters))
+                    }
                 }
             }
+        } yield (identity, action)
+
+        completeRequest(
+            queuedActivation = precheck flatMap {
+                case (identity, action) => activate(identity, action, context)
+            },
+            projectResultField = Option(context.path).filter(_.nonEmpty).map(_.split("/").filter(_.nonEmpty).toList),
+            responseType = responseType)
+    }
+
+    private def activate(identity: Identity, action: WhiskAction, context: Context, behalfOf: Option[Identity] = None)(
+        implicit transid: TransactionId): Future[(ActivationId, Option[WhiskActivation])] = {
+        // precedence order for parameters:
+        // package.params -> action.params -> query.params -> request.entity (body) -> augment arguments (namespace, path)
+        val invokeParams = if (action.hasFinalParamsAnnotation) {
+            context.requestParams -- action.parameters.immutableParameters
+        } else context.requestParams // in the absence of immutable annotations, return the request untouched
+
+        if (invokeParams.size == context.requestParams.size) {
```

Comment:
Is there a reason you used the `intersect` method proposed by me above, but not in this case?